src/AbstractApplication.php

```diff
@@ -62,6 +67,29 @@ public function __construct(Input $input = null, Registry $config = null)
 		$this->input = $input instanceof Input ? $input : new Input;
 		$this->config = $config instanceof Registry ? $config : new Registry;
 
+		// Create the container
+		$container = $this->getContainer();
+
+		// Set the input
+		$container->set('Joomla\\Input\\Input', $this->input, false, true);
+		$container->alias('Input', 'Joomla\\Input\\Input');
```

Comment:
The "namespaced" aliases aren't needed IMO.  It's fine to just have the FQCN and a shortcut key as an alias.